ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx

```diff
@@ -2,23 +2,78 @@ import {ErrorNotification, FormField, NotificationType, SlidingPanel} from 'argo
 import {services} from '../../../shared/services';
 import {ApplicationRetryOptions} from '../application-retry-options/application-retry-options';
-import {ApplicationManualSyncFlags, ApplicationSyncOptions, FORCE_WARNING, REPLACE_WARNING, SyncFlags} from '../application-sync-options/application-sync-options';
+import {ApplicationManualSyncFlags, ApplicationSyncOptions, FORCE_WARNING, SyncFlags} from '../application-sync-options/application-sync-options';
 import {ApplicationSelector} from '../../../shared/components';
+import {confirmSyncingAppOfApps} from '../utils';
 
 interface Progress {
     percentage: number;
     title: string;
 }
 
+const InstanceLabel = 'app.kubernetes.io/instance';
+
 export const ApplicationsSyncPanel = ({show, apps, hide}: {show: boolean; apps: models.Application[]; hide: () => void}) => {
     const [form, setForm] = React.useState<FormApi>(null);
     const [progress, setProgress] = React.useState<Progress>(null);
     const getSelectedApps = (params: any) => apps.filter((_, i) => params['app/' + i]);
     const [isPending, setPending] = React.useState(false);
+    const [appList, setAppList] = React.useState(apps);
+    const syncHandler = (currentForm: FormApi, ctx: ContextApis, applications: models.Application[]) => {
+        const formValues = currentForm.getFormState().values;
+        const replaceChecked = formValues.syncOptions?.includes('Replace=true');
+        const selectedApps = [];
+        const selectedAppOfApps: models.Application[] = [];
+        let containAppOfApps = false;
+
+        for (const key in formValues) {
+            if (key.startsWith('app/') && formValues[key]) {
+                selectedApps.push(applications[parseInt(key.slice(key.lastIndexOf('/') + 1), 10)]);
+            }
+        }
+
+        selectedApps.forEach(app => {
+            if (app.isAppOfAppsPattern) {
+                containAppOfApps = true;
+                selectedAppOfApps.push(app);
+            }
+        });
+
+        if (replaceChecked && containAppOfApps) {
+            confirmSyncingAppOfApps(selectedAppOfApps, ctx, currentForm).then(confirmed => {
+                setPending(confirmed ? true : false);
+            });
+        } else {
+            currentForm.submitForm(null);
+        }
+    };
+
+    React.useEffect(() => {
+        const appOfAppsList: string[] = [];
+        apps.forEach(app => {
+            const appName = app.metadata.labels && app.metadata.labels[InstanceLabel];
+            if (appName?.length > 0 && appOfAppsList.indexOf(appName) < 0) {
+                appOfAppsList.push(app.metadata.labels[InstanceLabel]);
+            }
+        });
+        const newAppList: models.Application[] = apps.map(app => {
+            const newApp = {...app, isAppOfAppsPattern: false};
+            if (appOfAppsList.indexOf(app.metadata.name) > -1) {
+                newApp.isAppOfAppsPattern = true;
+            }
+            return newApp;
+        });
```

Comment:
Because `apps` is the condition of `useEffect`, if the value of `apps` gets updated, it triggers lines inside `useEffect` method which will cause a execution loop.